### PR TITLE
OID4VCI Updates

### PR DESCRIPTION
**Pull request by nodh**

### CODE REVIEW

File: vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/SimpleAuthorizationService.kt [hunk 1]
```diff
@@ -145,16 +145,13 @@ class SimpleAuthorizationService(
...
-                if (!credentialSchemes.map { it.vcType }.contains(credentialIdentifier)) {
-                    throw OAuth2Exception(Errors.INVALID_GRANT)
-                        .also { Napier.w("token: client requested invalid credential identifier: $credentialIdentifier") }
-                }
-            }
-        }
+        // TODO work out mapping of credential identifiers in authorization details to schemes
+        val filteredAuthorizationDetails = params.authorizationDetails?.filter {
+            credentialSchemes.map { it.vcType }.contains(it.credentialConfigurationId) ||
+            credentialSchemes.map { it.sdJwtType }.contains(it.credentialConfigurationId)
```
Comment (acrusage-iaik) [inline vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/SimpleAuthorizationService.kt:151]: possible typo? Missing || at end of line, effectively making the first two lines useless

**Review by acrusage-iaik — CHANGES_REQUESTED** [review]
Possible logic error?

It also feels like the force unwrapping has gone a little out of control, but I couldn't think of a better way in the short term either

File: vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/WalletService.kt [hunk 2]
```diff
@@ -111,45 +139,101 @@ class WalletService(
         val clock: Clock = Clock.System,
     )
 
+    /**
+     * Pass in the URL provided by the Credential Issuer,
+     * which may contain a direct [CredentialOffer] or a URI pointing to it.
+     */
+    suspend fun parseCredentialOffer(input: String): KmmResult<CredentialOffer> = catching {
+        kotlin.runCatching {
```
Comment (JesusMcCloud) [inline vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/WalletService.kt:147]: `catching { runCatching`?

**Review by JesusMcCloud — CHANGES_REQUESTED** [review]
I'm honestly out of touch with the current state of vclib, but i did notice a creative pattern

File: vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/WalletService.kt (hunk 2 above)
Comment (nodh) [reply vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/WalletService.kt:147]: You're right, but imho changing this all the way through would pollute this PR.

Comment (JesusMcCloud) [reply vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/WalletService.kt]: all the way through is not the goal for this PR, but nesting it like that just seems off

**Review by acrusage-iaik — APPROVED** [review]

File: vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/WalletService.kt (hunk 2 above)
Comment (JesusMcCloud) [reply vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/WalletService.kt]: catching directly inside catchig still seems fishy

Comment (JesusMcCloud) [reply vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/WalletService.kt]: my bad. looking at the full functions helps. perfectly fine this way

**Review by JesusMcCloud — APPROVED** [review]

**Pull request merged**